### DISCUSSION TITLE
Support Cloud Shell VNETs in secondary regions

### DISCUSSION THREAD

**Discussion by maertendMSFT** [Azure/CloudShell#222] [Ideas]
Evaluating support for secondary regions.  More info on VNET support can be found in our [documentation](https://docs.microsoft.com/azure/cloud-shell/private-vnet#storage-requirements).

This change would add support for secondary regions where Azure Container Instances is present.

**Comment by DXPetti** [comment]
Currently, my customers cannot consume Cloud Shell for a number of reasons:

1) Compute is located out of the region
2) Public connectivity to resources is required (which CS VNets w/ Relay resolves)
3) Control around the session, specifically copy/paste and upload/download is not available.

Adding my support for this idea as it solves one problem

**Comment by rowanmoul** [comment]
Is this part of the "Improving the configuration options of the VNET scenario." bullet point from the 2023 road map? I would like to be able to use cloud shell to run things like terraform more securely, but I would like to keep the storage account in my own country for data residency reasons.

**Comment by mbifeld** [comment]
Cloud Shell in a VNet now supports secondary regions.
